File: ultimate/HttpRequest.java
package ultimate;

public class HttpRequest {

    private String message;
    private User user;

    public HttpRequest(User user, String message) {
        this.user = user;
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    public User getUser() {
        return user;
    }
}

File: typescript/patterns/ultimate-design-pattern-challenge/http-request.ts
class HttpRequest {
  private message: string;
  private user: User;

  constructor(message: string, user: User) {
    this.message = message;
    this.user = user;
  }

  getMessage(): string {
    return this.message;
  }

  getUser(): User {
    return this.user;
  }
}
